
uint32 x_id
uint32 y_id
uint32 loc_x
uint32 loc_y